### PR TITLE
add SSP module interface

### PR DESCRIPTION
**Pull request by chrisroadmap**
Addresses #43

- [x] Tests added
- [x] Documentation added (where applicable)
- [x] Example added (either to an existing notebook or as a new notebook, where applicable)
- [x] Description in ``CHANGELOG.rst`` added
- [x] Refactor RCP interface

### CODE REVIEW

**Comment by chrisroadmap** [comment]
wait for #102 (a good test would be if this branch fails coverage without tests being added)

**Comment by chrisroadmap** [comment]
I shouldn't be happy that coverage has gone down, but I am here, because it means the CI is working

**Comment by znicholls** [comment]
Not sure if this still a draft, but I'm hoping that you're planning on removing all the duplicate code? It must be possible to get the same interfaces without all this duplication (tidying up the RCPs at the same time is probably not a bad idea either).

**Comment by chrisroadmap** [comment]
Any ideas how to do this? I know what I'm doing currently is very unpythonic but it's been around a while.

Just want this to work:

```
from fair.SSPs import rcp119
from fair.forward import fair_scm

...

fair_scm(emissions=ssp119.Emissions.emissions)
```

What I need to figure out is how to put all of the duplicated stuff in one module, import the particular SSP, then get the SSP module to communicate with the shared module (give it the specific SSP to look up from the emissions data file, then put the results in the Emissions class) and be available to the main interface using the original SSP name.

**Comment by znicholls** [comment]
How about something like

```python
# shared module
class Emissions:
    def __init__(self, emissions):
        self.emissions = emissions
        # etc. etc. 

def load_emms_data(fp):
    # do the loading, return something which has the attributes you want
    # e.g.
    loaded = pd.read_csv(fp)
    out = Emissions(loaded)

    return out
```

```python
# SSPs module
class ssp119:
    Emissions = load_emms_data("path_to_ssp119.txt")

# then you should be able to do
ssp119.Emissions.emissions  # etc.
```

**Comment by chrisroadmap** [comment]
> Not sure if this still a draft, but I'm hoping that you're planning on removing all the duplicate code? It must be possible to get the same interfaces without all this duplication (tidying up the RCPs at the same time is probably not a bad idea either).

- [ ] Refactor RCP interface

File: fair/SSPs/__init__.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+emissions_filename = os.path.join(
+    os.path.dirname(__file__), 'data/rcmip-emissions-annual-means-5-1-0-ssp-only.csv')
+concentrations_filename = os.path.join(
+    os.path.dirname(__file__), 'data/rcmip-concentrations-annual-means-5-1-0-ssp-only.csv')
+forcing_filename = os.path.join(
+    os.path.dirname(__file__), 'data/rcmip-radiative-forcing-annual-means-5-1-0-ssp-only.csv')
+
+
+# I know we can refactor this but can't think immediately how
+class ssp119:
```
Comment (znicholls) [inline fair/SSPs/__init__.py]: Something like the below would probably work

```python
class FaIRHelper:
    pass 


def get_ssp(efn, cfn, ffn, ssp):
    emissions = load_emissions_data(efn, ssp)
    # same for conc and forcing
    out = FaIRHelper()
    out.Emissions = emissions
    out.Concentrations = concentrations
    # etc.
    return out


ssp119 = get_ssp(emissions_filename, concentrations_filename, forcing_filename,"ssp119")
ssp126 = get_ssp(emissions_filename, concentrations_filename, forcing_filename,"ssp126")
# etc.
```

**Comment by chrisroadmap** [comment]
@znicholls OK let me know what you think. I could probably avoid duplication in the `__init__.py` files but can't figure out exactly how to do it. Coverage decreased because we removed code! (this won't stop me merging as still over 95%).

**Comment by znicholls** [comment]
> Coverage decreased because we removed code! (this won't stop me merging as still over 95%).

Nice, only one suggestion in the comment above

**Pull request merged**